### PR TITLE
Ft chat application 187300175

### PR DESCRIPTION
**Pull request by Ruthuwamahoro**
#### What does this PR do?

-creation of chat application

#### Description of Task to be completed?

- Allow only authenticated people to use our chat application
- Sent message in real time application
- Storing all messages in database

#### How should this be manually tested?

- set up the env according to your localhost database credentials
- use the api in the index.ts in routers to connect to the database
- run npm run install to install the necessary dependencies
- use npm run dev to connect to the database

#### What are the relevant pivotal tracker/Trello stories?

- Acceptance Criteria

GIVEN the user is online
WHEN they open the chat tab
THEN they should be able to read all of the past chats

GIVEN the user is online
WHEN they open the chat tab
THEN they should be able to post a public chat

GIVENthe user
WHEN they post a chat
THEN it should be able to be echoed to all online users in realtime.

- id of the project
  #187300175

### CODE REVIEW

**Review by fab-ryan — CHANGES_REQUESTED** [review]
work on that only

File: package.json [hunk 1]
```diff
@@ -106,6 +107,7 @@
     "joi": "^17.12.2",
     "jsonwebtoken": "^9.0.2",
     "multer": "^1.4.5-lts.1",
+    "mysql2": "^3.9.4",
```
[the remark this answers is not in the document]
Comment (Ruthuwamahoro) [reply package.json]: solved

[the remark this answers is not in the document]
Comment (Mukunzijames) [reply package.json:110]: also squash commit you have 4 commits

**Review by Mukunzijames — APPROVED** [review]

File: src/server.ts [hunk 2]
```diff
@@ -1,10 +1,73 @@
...
+        senderId,
+        content,
+        readStatus,
+      });
+    }
+  });
+  socket.on('typing', isTyping => {
+    socket.broadcast.emit('typing', isTyping);
+  });
+});
```
Comment (fab-ryan) [inline src/server.ts]: wrap this into single function then call that function here

**Review by fab-ryan — CHANGES_REQUESTED** [review]
wrapping the functionality of sockets

**Review by fab-ryan — APPROVED** [review]

**Review by RWEMAREMY — APPROVED** [review]
good job

**Review by Nadinefiona — APPROVED** [review]
Good Work 👏

**Pull request merged**